ui/app/routes/api/tensorzero/inference.utils.tsx

```diff
@@ -217,37 +217,38 @@ function tensorZeroStoredContentToInputContent(
   content: TensorZeroStoredInputMessageContent,
 ): InputMessageContent {
   switch (content.type) {
-    case "text":
-      return {
-        type: "text",
-        value: content.text,
-      };
-    case "template":
     case "raw_text":
+    case "template":
+    case "text":
+    case "thought":
+    case "tool_result":
+    case "unknown":
       return content;
     case "tool_call":
-      return {
-        type: "tool_call",
-        id: content.id,
-        name: content.name,
-        arguments: content.arguments,
-      };
-    case "tool_result":
-      return {
-        type: "tool_result",
-        id: content.id,
-        name: content.name,
-        result: content.result,
-      };
-    case "thought":
-      return {
-        type: "thought",
-        text: content.text,
-        _internal_provider_type: content._internal_provider_type,
-        signature: content.signature,
-      };
+      if ("raw_arguments" in content) {
```

Comment:
Unfortunately same issue - we can't distinguish the user-created vs. backend-stored versions of tool calls anymore.